pkg/rpc/server/oracle.go

```diff
@@ -0,0 +1,68 @@
+package server
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"time"
+
+	"github.com/nspcc-dev/neo-go/pkg/config/netmode"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/client"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/request"
+)
+
+type rpcBroascaster struct {
+	magic   netmode.Magic
+	clients map[string]*client.Client
+}
+
+func newRPCBroadcaster(magic netmode.Magic, addrs []string) *rpcBroascaster {
+	r := &rpcBroascaster{magic: magic}
+	for i := range addrs {
+		// We ignore error as not every node can be available on startup.
+		r.clients[addrs[i]], _ = client.New(context.Background(), addrs[i], client.Options{
+			DialTimeout:    time.Millisecond * 500,
+			RequestTimeout: time.Millisecond * 500,
+			Network:        r.magic,
+		})
+	}
+	return r
+}
+
+func (r *rpcBroascaster) broadcastResponse(priv *keys.PrivateKey, resp *transaction.OracleResponse, txSig []byte) {
+	pub := priv.PublicKey()
+	data := getOracleMessage(pub.Bytes(), resp.ID, txSig)
+	msgSig := priv.Sign(data)
+	params := request.NewRawParams(
+		hex.EncodeToString(pub.Bytes()),
+		resp.ID,
+		hex.EncodeToString(txSig),
+		hex.EncodeToString(msgSig),
+	)
+	for addr, c := range r.clients {
+		if c == nil {
+			var err error
+			c, err = client.New(context.Background(), addr, client.Options{
+				DialTimeout:    time.Millisecond * 500,
+				RequestTimeout: time.Millisecond * 500,
```

Comment:
Arbitrary values.

Reply:
Reasonable time to process request. We may make them 4 times bigger, but they will remain arbitrary.

Reply:
We can make them be the same as for the oracle request itself (and configurable this way).

Reply:
But these are completely unrelated. We may provide this in config, though.

Reply:
Well, both are HTTP requests, so using one setting for both is fine. But making it a separate option is also OK.